pkg/ebpftracer/tracer.go

```diff
@@ -195,45 +197,16 @@ func (t *Tracer) GetEventName(id events.ID) string {
 	return ""
 }
 
-func (t *Tracer) signalReadLoop(ctx context.Context) error {
-	eventsReader, err := perf.NewReader(t.module.objects.Signals, t.cfg.EventsPerCPUBuffer)
-	if err != nil {
-		return err
-	}
-	defer eventsReader.Close()
-
-	for {
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		default:
-		}
-		record, err := eventsReader.Read()
-		if err != nil {
-			if t.cfg.DebugEnabled {
-				t.log.Warnf("reading signals: %v", err)
-			}
-			continue
-		}
-		if record.LostSamples > 0 {
-			t.log.Warnf("lost %d signals", record.LostSamples)
-			metrics.AgentKernelLostEventsTotal.Add(float64(record.LostSamples))
-			continue
-		}
-		metrics.AgentPulledEventsTotal.Inc()
-
-		if err := t.decodeAndHandleSignal(ctx, record.RawSample); err != nil {
-			if t.cfg.DebugEnabled || errors.Is(err, ErrPanic) {
-				t.log.Errorf("decoding signal: %v", err)
-			}
-			metrics.AgentDecodeEventErrorsTotal.Inc()
-			continue
-		}
-	}
+func (t *Tracer) signalEventsReadLoop(ctx context.Context) error {
+	return t.runPerfBufReaderLoop(ctx, t.module.objects.Events)
```

Comment:
Do I understand correctly that now we will have single exec event but it will come from signal per buff?

Reply:
Exactly, this completely eliminates the overhead of sending two events and instead just moves some events over to the new "high prio" perf buf.